js/src/elements/tab-container/element.ts

```diff
@@ -0,0 +1,473 @@
+			tab => tab instanceof HTMLElement && tab.closest(this.tagName) === this
+		);
+	}
+
+	get #next(): HTMLButtonElement {
+		return this.querySelector<HTMLButtonElement>('[data-tab-container-nav="next"]')!;
+	}
+	get #prev(): HTMLButtonElement {
+		return this.querySelector<HTMLButtonElement>('[data-tab-container-nav="previous"]')!;
+	}
+
+	get activeTab() {
+		return this.#tabs[this.selectedTabIndex];
+	}
+
+	get activePanel() {
+		return this.#panelSlot.assignedNodes()[0] as HTMLElement;
+	}
+
+	get vertical(): boolean {
+		return this.#tabList?.getAttribute('aria-orientation') === 'vertical';
+	}
+
+	set vertical(isVertical: boolean) {
+		const tabList = this.#tabList;
+		if (tabList && isVertical) {
+			tabList.setAttribute('aria-orientation', 'vertical');
+		} else {
+			tabList.setAttribute('aria-orientation', 'horizontal');
+		}
+	}
+
+	#setupComplete = false;
+	#internals!: ElementInternals | null;
+	connectedCallback(): void {
+		this.#internals ||= this.attachInternals ? this.attachInternals() : null;
+		const shadowRoot =
+			this.shadowRoot || this.attachShadow({ mode: 'open', slotAssignment: 'manual' });
+		const tabListContainer = document.createElement('slot');
+		tabListContainer.style.display = 'flex';
+		tabListContainer.setAttribute('part', 'tablist-wrapper');
+		tabListContainer.setAttribute('name', 'tablist-wrapper');
+		const tabListTabWrapper = document.createElement('div');
+		tabListTabWrapper.setAttribute('part', 'tablist-tab-wrapper');
+		tabListTabWrapper.setAttribute('name', 'tablist-tab-wrapper');
+		const tabListSlot = document.createElement('slot');
+		tabListSlot.setAttribute('part', 'tablist');
+		tabListSlot.setAttribute('name', 'tablist');
+		tabListTabWrapper.append(tabListSlot);
+		const panelSlot = document.createElement('slot');
+		panelSlot.setAttribute('part', 'panel');
+		panelSlot.setAttribute('name', 'panel');
+		const beforeTabSlot = document.createElement('slot');
+		beforeTabSlot.setAttribute('part', 'before-tabs');
+		beforeTabSlot.setAttribute('name', 'before-tabs');
+		const afterTabSlot = document.createElement('slot');
+		afterTabSlot.setAttribute('part', 'after-tabs');
+		afterTabSlot.setAttribute('name', 'after-tabs');
+		tabListContainer.append(beforeTabSlot, tabListTabWrapper, afterTabSlot);
+		const afterSlot = document.createElement('slot');
+		afterSlot.setAttribute('part', 'after-panels');
+		afterSlot.setAttribute('name', 'after-panels');
+		shadowRoot.replaceChildren(tabListContainer, panelSlot, afterSlot);
+
+		if (this.#internals && 'role' in this.#internals) {
+			this.#internals.role = 'presentation';
+		} else {
+			this.setAttribute('role', 'presentation');
+		}
+
+		this.addEventListener('keydown', this);
+		this.addEventListener('click', this);
+
+		this.selectTab(-1);
+
+		this.addEventListener('tab-container-change', this.#scrollToTab);
+		this.#next?.addEventListener('click', this.#handleNavClick.bind(this));
```

Comment:
Not that these should be called on `this.` not on the buttons themselves. The suggested change I wrote earlier allows for either in one function without bind.

Reply:
You're right.  I got too fancy and my mental modal of the click listener was too specific.  Updated.